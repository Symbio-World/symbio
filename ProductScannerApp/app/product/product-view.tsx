import React from 'react'
import { Image, Text, ScrollView } from 'react-native'
import { t } from 'react-native-tailwindcss'
import { ProductData } from 'fetcher-core'

type Props = Omit<ProductData, 'links'>

export const ProductView: React.FC<Props> = props => (
  <ScrollView style={[t.bgWhite, t.flex1]}>
    <Image style={[t.h56, t.bgGray200]} source={{ uri: props.image }} resizeMode='contain' />
    <Text style={[t.textLg, t.textCenter]}>{props.name}</Text>
    <Text>{props.description}</Text>
    <Text>{props.ingredients}</Text>
    <Text>{props.allergens}</Text>
    <Text>{props.origin}</Text>
  </ScrollView>
)
