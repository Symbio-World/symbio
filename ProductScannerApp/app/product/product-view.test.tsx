import 'react-native'
import React from 'react'
import { render } from 'react-native-testing-library'
import { ProductView } from './product-view'

describe('ProductView', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<ProductView />)
    expect(toJSON()).toMatchSnapshot()
  })
})
