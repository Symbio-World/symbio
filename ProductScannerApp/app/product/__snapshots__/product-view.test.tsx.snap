// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductView renders correctly 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View>
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "height": 224,
          },
          Object {
            "backgroundColor": "#edf2f7",
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "fontSize": 18,
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    />
    <Text />
    <Text />
    <Text />
    <Text />
  </View>
</RCTScrollView>
`;
